src/txn/txn_ckpt.c

```diff
@@ -1440,8 +1451,11 @@ __checkpoint_tree(
 	 * If the checkpoint didn't complete successfully, make sure the
 	 * tree is marked dirty.
 	 */
-	if (ret != 0 && !btree->modified && was_modified)
+	if (ret != 0 && !btree->modified && was_modified) {
 		btree->modified = 1;
+		if (S2C(session)->modified == 0)
+			S2C(session)->modified = 1;
+	}
```

Comment:
As well as changing `S2C(session)->modified` here, please update the line above to be `btree->modified = true`.